### ISSUE TITLE
Undebates Header Bar

### ISSUE THREAD

**Issue by ddfridley** [EnCiv/undebate-ssp#22]
<UndebatesHeaderBar
>

- [x] Magnifying Glass / Search - functionality to be implemented later
- [x] Book/Instructions functionality to be implemented later
- [ ]  Create New takes us to the create undebate page
- [x] User Setting / button have minimal data for now - starting with logout

Here is a link to figma: https://www.figma.com/file/yXrcMvg5EIwdLkHHxIdILh/Undebate?node-id=326%3A5505

![image](https://user-images.githubusercontent.com/3317487/131608431-d0de8ea6-093a-4581-8c9c-5a52be9c5744.png)

**Comment by ddfridley** [comment]
@Bob160 Hi - we haven't talked in a while. Did you get the repo working?  I'm going to unassign this task for now in case some has time to work on it. But feel free to assign it to yourself if you want it back.

**Comment by alex-808** [comment]
Hey @ddfridley I have been working on this component and have a couple of questions. I've pushed my branch so you can check out the current state of it.

1. I'm assuming that the 'Create New', 'Instructions' and 'Search' buttons are going to make requests to the server. Is there any way you'd like me to mock that up (API request functions it would accept as props maybe) or should I just leave them as buttons for now?

2. At the moment all I'm pulling out the props is a 'userId' as is specified in the Figma. Anything else it should accept such as 'style' and 'className'?

3. I am having issues with responsiveness in this component. The problem originates from the hover effect specified in the Figma which expands pretty substantially and would just barely fit on screen @ 320px width. If you look at the component as it is now you can see some of the steps I took to combat this issue but I'm wondering if it might be a good idea for Kamal to draft a mobile version of the header as he can probably do it better than I can.

If there's anything else you see in the code, let me know but that's all I had for now. Thanks

**Comment by ddfridley** [comment]
@alex-808 I have been working on the storybook problem all day and it's messed up my environment and It's going to be some work to get it back. 

Yes - it would be good to accept style and className

1) I can provide you more detail on Create New.

For 2, look for a prop called user, that will be {id: 'unique-id', email: string}, or it will be undefined if not logged in yet. We don't currently have the concept of a use image.  If we did, we would put it there. For now, we need to find a default image.

For 3, just make sure you are using rem and not px.  For this portal, working on a smartphone is not a high priority. 

For Instructions, if you merge with main there will be a new component instruction-button.  Use that. It just returns the icon for now, but in the future we will put more logic in there.  Also remember to take the instruction svg out of assets so we don't have conflicts.

**Comment by ddfridley** [comment]
to log out, navigate to /sign/out

**Comment by alex-808** [comment]
@ddfridley awesome, all seems clear. Thanks David

**Comment by ddfridley** [comment]
@al

**Comment by ddfridley** [comment]
@alex-808 
I saw one thing after I merged the PR.
There is:
`"@media screen and (max-width: 840px)"`
It violates the no px rule.  When you have been working with displays for a long time, you will see that sizes change. Eventually, in some environment on some display, 840px will be the wrong answer.

What seems more appropriate here is:
'@media (orientation: portrait)'
Which says why (because you are in portrait mode) you are doing this. 

Or, it may be that you would rather say something in rem (and I don't know if media queries accept that) because you are saying that if the screen is less than the width of what you are trying to put there (which is all specified in rem in some way), then you should go into the stacked layout".  

Or, it's overkill and you shouldn't invest the time in this for this case - render the header and then in useLayoutEffect check the width after it is rendered, and if it's bigger than the screen width, switch to the column layout. 

It's a minor thing but maybe it will help you in the future. Can you make a quick change in the branch and push it again. Thx.

**Comment by alex-808** [comment]
@ddfridley good catch, wrote that out of pure habit. It turns out media queries do accept rems so I implemented that. I've pushed it to the branch. Let me know if it needs anything else.

I originally added the media query purely to prevent the icons crowding and looking nasty when the user icon hover effect expanded so the '@media (orientation: portrait)' solution wasn't quite what I was shooting for but I'm glad to learn about it.

Thanks for sharing an example use case for useLayoutEffect as well. I haven't used that hook yet but now I'll look into it.
